### ISSUE TITLE
Server platform doesn't build

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#8361]
**Operating system or device - Godot version:**
Linux, Godot master/HEAD (faee2fb )

**Issue description:**
The server platform needs fixed. It's true. It's gonna be great.

Two things need to be fixed:
- The AudioServer setup hasn't been fully adapted to the new API. It's easy to fix, just remove that obsolete stuff, and related lines affecting/declaring those members: https://github.com/godotengine/godot/blob/master/platform/server/os_server.cpp#L72-L78
- Then it needs to be adapted for the GLES3 renderer, and here I have less experience to tell what needs to be done :) It should likely be compared with the implementation in 2.1 to see how it worked.

**Steps to reproduce:**
- `scons p=server`

**Comment by Hinsbart** [comment]
I think we should also take this opportunity to finally get rid of the designated server platform and provide a server=yes compilation flag that also works on non-unix platforms.
See https://github.com/godotengine/godot/issues/6059#issuecomment-238195785

I actually did already get started on abstracting out the windowing system from the platform code back in january/february. IIRC I've got it working for x11, then wanted to integrate the old server code but noticed that I'd have to write a DummyRasterizer for the new gles3 renderer (HUGE task). That's when I've put the project on hold and forgot about it...
So, thanks for the reminder :P

**TL;DR** Looking for volunteers to bring back the dummy rasterizer on gles3 :)

**Comment by GrandFunkR** [comment]
what is the reason to have AudioServer and GLES3 in p=server?
not enough to have a guiless mode and no audio in  platform server?

**Comment by Hinsbart** [comment]
@GrandFunkR yes, we don't want to have gles3 rendering or audio in the server. But the rasterizer and audio server also handle behind-the-scenes resource management that the engine needs in any case, regardless of whether there will be any rendering or audio processing or not. That's why we have to bring back the dummy drivers.

**Comment by GrandFunkR** [comment]
Is this a feature that will be implemented in godot 3.0?

**Comment by reduz** [comment]
kicking to 3.1, apologies.

**Comment by GrandFunkR** [comment]
It is there a way to run gdscript on a Linux server with no audio card and no X11 without server compile?

**Comment by akien-mga** [comment]
> It is there a way to run gdscript on a Linux server with no audio card and no X11 without server compile?

Nope. But 3.1 should come soon enough for those who need this in their workflow.

**Comment by fire** [comment]
@Faless Want to rebase and make a pull request for https://github.com/Faless/godot/commits/server_again

**Comment by Faless** [comment]
> @Faless Want to rebase and make a pull request for https://github.com/Faless/godot/commits/server_again

I'm not sure that's how @reduz and other devs want it to be implemented.
The thing needs further discussion so I find it unlikely that it will be merged for 3.0

**Comment by Faless** [comment]
@fire anyway, I've rebased, and fixed it to match the updated engine API.

**Comment by fire** [comment]
Thank you, I wanted to compile it and think of the design with a working sample.

**Comment by fire** [comment]
Do you know if it's difficult to have a Windows compatible server for testing? My preferred server platform is Linux..

**Comment by fire** [comment]
Noticed a problem, the server assumes linux, while theoretically it could be mac, linux or windows. Any thoughts of how to solve this configuration issue?

**Comment by akien-mga** [comment]
It's not a configuration issue, `server` is a Linux platform. See #11389 for the discussion on a cross-platform server platform/build option.

**Comment by fire** [comment]
Noted that if the platform has no alsa or pulseaudio it fails.

**Comment by fire** [comment]
@Faless Can you think of a different name than dummy? It feels insentive to some people.

**Comment by Faless** [comment]
I fail to see how the proper English word for "model" and "not real" can be insensitive.
https://dictionary.cambridge.org/dictionary/english/dummy

I think that if one feels offended by a word used in the proper context and meaning, well... that person should reconsider its priorities in life.

I would suggest that person to start believing in himself instead of shielding from the world, which is useless and actually detrimental.

**Comment by Faless** [comment]
Fixed by #16653
